pkg/gui/controllers/list_controller.go

```diff
@@ -82,6 +82,12 @@ func (self *ListController) handleLineChange(change int) error {
 	// doing this check so that if we're holding the up key at the start of the list
 	// we're not constantly re-rendering the main view.
 	if before != after {
+		if change == -1 {
+			checkScrollUp(self.context.GetViewTrait(), self.c.UserConfig.Gui.ScrollOffMargin, before, after)
```

Comment:
In the other branch we will have to call `ModelToViewIndex` on `before` and `after` to make it work correctly for lists that have section headers.